rpc/src/lib.rs

```diff
@@ -1,2 +1,192 @@
+                        .header
+                        .mix_hash
+                        .context("target block is missing field `mix_hash`")?
+                        .compat(),
+                    block_gaslimit: target_block.header.gas_limit.into(),
+                    block_chain_id: chain_id.into(),
+                    block_base_fee: target_block
+                        .header
+                        .base_fee_per_gas
+                        .context("target block is missing field `base_fee_per_gas`")?
+                        .into(),
+                    block_gas_used: target_block.header.gas_used.into(),
+                    block_bloom: target_block.header.logs_bloom.compat(),
+                },
+                b_hashes: BlockHashes {
+                    prev_hashes: prev_hashes.map(|it| it.compat()).into(),
+                    cur_hash: target_block
+                        .header
+                        .hash
+                        .context("target block is missing field `hash`")?
+                        .compat(),
+                },
+                withdrawals: target_block
+                    .withdrawals
+                    .into_iter()
+                    .flatten()
+                    .map(
+                        |Withdrawal {
+                             address, amount, ..
+                         }| { (address.compat(), amount.into()) },
+                    )
+                    .collect(),
+            },
+            checkpoint_state_trie_root: checkpoint_state_trie_root.compat(),
+        },
+    })
+}
+
+trait Compat<Out> {
+    fn compat(self) -> Out;
+}
+
+impl Compat<__compat_primitive_types::H160> for alloy::primitives::Address {
+    fn compat(self) -> __compat_primitive_types::H160 {
+        let alloy::primitives::Address(alloy::primitives::FixedBytes(arr)) = self;
+        __compat_primitive_types::H160(arr)
+    }
+}
+
+impl Compat<__compat_primitive_types::H256> for alloy::primitives::B256 {
+    fn compat(self) -> __compat_primitive_types::H256 {
+        let alloy::primitives::FixedBytes(arr) = self;
+        __compat_primitive_types::H256(arr)
+    }
+}
+
+impl Compat<[__compat_primitive_types::U256; 8]> for alloy::primitives::Bloom {
+    fn compat(self) -> [__compat_primitive_types::U256; 8] {
+        let alloy::primitives::Bloom(alloy::primitives::FixedBytes(src)) = self;
+        // have      u8 * 256
+        // want    U256 * 8
+        // (no unsafe, no unstable)
+        let mut chunks = src.chunks_exact(32);
+        let dst = core::array::from_fn(|_ix| {
+            // This is a bit spicy because we're going from an uninterpeted array of bytes
+            // to wide integers, but we trust this `From` impl to do the right thing
+            __compat_primitive_types::U256::from(
+                <[u8; 32]>::try_from(chunks.next().unwrap()).unwrap(),
+            )
+        });
+        assert_eq!(chunks.len(), 0);
+        dst
+    }
+}
+
+#[test]
+fn bloom() {
```

Comment:
I'm going to gently push for more tests here, I guess that's not a massive surprise :)